src/app/commands/cmd_layer_visibility.cpp

```diff
@@ -45,10 +47,16 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
 bool LayerVisibilityCommand::onChecked(Context* context)
 {
   const ContextReader reader(context);
-  const SelectedLayers& selLayers = reader.site()->selectedLayers();
+  SelectedLayers selLayers;
+  auto range = App::instance()->timeline()->range();
+  if (range.enabled()) {
+    selLayers = range.selectedLayers();
+  }
+  else {
+    selLayers.insert(const_cast<Layer*>(reader.layer()));
```

Comment:
I felt a bit dirty const_casting this. I looked for examples of usage with reader.layer() in other commands following the pattern you suggested and didn't find any. I can modify this code to use logic that doesn't cast away the const if you like.